xml/System.IO.Pipelines/PipeOptions.xml

```diff
@@ -45,14 +45,14 @@
         <Parameter Name="useSynchronizationContext" Type="System.Boolean" />
       </Parameters>
       <Docs>
-        <param name="pool">To be added.</param>
-        <param name="readerScheduler">To be added.</param>
-        <param name="writerScheduler">To be added.</param>
-        <param name="pauseWriterThreshold">To be added.</param>
-        <param name="resumeWriterThreshold">To be added.</param>
-        <param name="minimumSegmentSize">To be added.</param>
-        <param name="useSynchronizationContext">To be added.</param>
-        <summary>Creates a new instance of <see cref="T:System.IO.Pipelines.PipeOptions" />.</summary>
+        <param name="pool">The pool of memory blocks to be used for buffer management.</param>
+        <param name="readerScheduler">The <see cref="T:System.IO.Pipelines.PipeScheduler" /> to be used to execute <see cref="T:System.IO.Pipelines.PipeReader" /> callbacks and async continuations.</param>
+        <param name="writerScheduler">The <see cref="T:System.IO.Pipelines.PipeScheduler" /> used to execute <see cref="T:System.IO.Pipelines.PipeWriter" /> callbacks and async continuations.</param>
+        <param name="pauseWriterThreshold">The number of bytes in the <see cref="T:System.IO.Pipelines.Pipe" /> before <see cref="M:System.IO.Pipelines.PipeWriter.FlushAsync(System.Threading.CancellationToken)" /> starts blocking. A negative value prevents <see cref="M:System.IO.Pipelines.PipeWriter.FlushAsync(System.Threading.CancellationToken)" /> from ever blocking, effectively making the number of bytes in the <see cref="T:System.IO.Pipelines.Pipe" /> unlimited.</param>
+        <param name="resumeWriterThreshold">The number of bytes in the <see cref="T:System.IO.Pipelines.Pipe" /> when <see cref="M:System.IO.Pipelines.PipeWriter.FlushAsync(System.Threading.CancellationToken)" /> stops blocking.</param>
```

Comment:
@halter73 you wrote:

> _Personally, I would use "async continuations" instead of "callbacks" everywhere "callbacks" are referenced. I would also make sure everywhere thresholds are referenced, we communicate that setting them to a negative value disables them._

But I'm not entirely sure how to explain it for `resumeWriterThreshold`. Would you mind helping here?

Reply:
This one is a little weird because technically you don't need to disable `resumeWriterThreshold` if you've already disabled the `pauseWriterThreshold`. And if the `pauseWriterThreshold` *isn't* disabled, it would be silly to disable the `resumeWriterThreshold`. So keeping this as-is and not mentioning what happens when you set the  `resumeWriterThreshold` negative is probably fine.